vars/multicloudDeploy.groovy

```diff
@@ -114,17 +139,21 @@ void call(parameters = [:]) {
             error "Deployment skipped because no targets defined!"
         }
 
-        echo "Executing deployments"
-        if (config.parallelExecution) {
-            echo "Executing deployments in parallel"
-            parallel deployments
-        } else {
-            echo "Executing deployments in sequence"
-            def closuresToRun = deployments.values().asList()
-            Collections.shuffle(closuresToRun) // Shuffle the list so no one tries to rely on the order of execution
-            for (int i = 0; i < closuresToRun.size(); i++) {
-                (closuresToRun[i] as Closure)()
-            }
+        runClosures(config, deployments, "deployments")
+
+    }
+}
+
+def runClosures(Map config, Map toRun, String label = "closures") {
+    echo "Executing $label"
+    if (config.parallelExecution) {
```

Comment:
[curiosity] What is the reason for making parallelism configurable. Are there cases where we expect issue with parallel execution? By the way: I understand that this was configurable already before, from that point of view my question is not fully appropriate here in that PR ...

Reply:
It can break blue ocean. Blue ocean cannot display two levels of parallelity. So if you have a parrallel stage which has a parallel block the pipeline on blue ocean looks strange but it still works.

Reply:
Yes, I've checked if blue ocean has improved in those cases, but it has not. It looks really confusing, this is why we did not enable it by default.